src/handleApps.test.ts

```diff
@@ -174,6 +175,49 @@ describe(`handleApps`, () => {
       "public, max-age=31536000, immutable"
     );
   });
+
+  it(`can retrieve a file from s3`, async () => {
+    const orgSettings: RecursivePartial<OrgSettings> = {
+      orgExists: true,
+      staticFiles: {
+        microfrontendProxy: {
+          environments: {
+            __main__: {
+              useFoundryHosting: false,
+              customHost: "s3://example",
+            },
+          },
+        },
+      },
+    };
+
+    (global.MAIN_KV as MockCloudflareKV).mockKv({
+      "org-settings-walmart": orgSettings,
+    });
+
+    sendMock.mockReturnValueOnce(
+      Promise.resolve({
+        Body: "var a = 1",
+        ContentType: "application/javascript",
+      })
+    );
+
+    response = await handleApps(
+      new Request("https://cdn.baseplate.cloud/walmart/apps/example.js"),
+      {
+        orgKey: "walmart",
+        customerEnv: "__main__",
+        pathParts: ["example.js"],
+      }
+    );
+    let responseBody = await response.text();
+
+    expect(response.ok).toBe(true);
+    expect(response.headers.get("content-type")).toEqual(
+      "application/javascript"
+    );
+    expect(responseBody).toEqual("var a = 1");
```

Comment:
Let's add some more assertions to verify the logic that's extracting the s3 bucket and key name:

```ts
expect(sendMock).toHaveBeenCalled();
expect(sendMock.mock.calls[0][0].Bucket).toEqual("example");
expect(sendMock.mock.calls[0][0].Key).toEqual("example.js");
```

Reply:
`sendMock.mock.calls[0][0].Bucket` is undefined. A console.log of `sendMock.mock.calls[0][0]` prints `GetObjectCommand {}` to the console. I'm not really sure how to get what we want out of this.

Reply:
Ah this is due to our mock of AWS client s3. See https://github.com/JustUtahCoders/baseplate-cloudflare-worker/pull/35/files#diff-5fdff6ec127891a829e02462bd611919cc6880ad5bf8699d84807529956767b3R8

Change that line to the following:

```ts
export class GetObjectCommand {
  constructor(opts) {
    this.Bucket = opts.Bucket;
    this.Key = opts.Key;
  }
}
```

Reply:
Oh that makes sense, thanks.